lib/call-stack-to-nix.nix

```diff
@@ -6,7 +6,11 @@
  * see also `call-cabal-project-to-nix`!
  */
 { runCommand, nix-tools, pkgs, mkCacheFile, materialize }:
-{ src, stackYaml ? null, ignorePackageYaml ? false, cache ? null
+{ name          ? src.name or null # optional name for better error messages
+, src
+, stackYaml ? null
+, ignorePackageYaml ? false
+, cache ? null
 , stack-sha256 ? null
```

Comment:
Add a `name` arg like the one in `call-cabal-project-to-nix.nix`.  I needed this to track down the unwanted evals.